clang/lib/Driver/Driver.cpp

```diff
@@ -4134,9 +4063,33 @@ class OffloadingActionBuilder final {
           // FPGA aoco does not go through the link, everything else does.
           if (Input->getType() == types::TY_FPGA_AOCO)
             DeviceLibObjects.push_back(Input);
-          else
+          // FPGA aocr/aocx does not go through the link and is passed
+          // directly to the backend compilation step (aocr) or wrapped (aocx)
+          else if (types::isFPGA(Input->getType())) {
+            Action *AOCAction;
+            constexpr char COL_CODE[] = "Code";
+            constexpr char COL_ZERO[] = "0";
+            if (Input->getType() == types::TY_FPGA_AOCR)
+              // Generate AOCX/AOCR
```

Comment:
Am I right to conclude that an `.aocr` can only be processed like this?
```suggestion
              // Generate AOCX
```

Reply:
There is the possible path of `.aocr` to `.aocr`.  Not clear what that means, but the behavior is accepted.